chombot-common/src/lib.rs

```diff
@@ -27,20 +29,15 @@ pub trait ChombotPoiseUserData: Sync {
 
 pub type ChombotPoiseContext<'a, T> = Context<'a, T, anyhow::Error>;
 
-pub fn start_tournaments_watcher(
-    tournaments_watcher_channel_id: Option<u64>,
+pub fn start_tournaments_watcher<T: TournamentWatcherChannelListProvider + 'static>(
+    channel_list_provider: T,
     ctx: SerenityContext,
 ) {
     const MESSAGE_PREFIX: &str =
```

Comment:
Is there a specific reason for this text to be provided by the user? It seems like the structure is being used for one reason only anyway, so why not specify it as const inside it?

Reply:
Indeed, that doesn't sound like a very bad idea. Perhaps I wanted this to be customizable by the users, but I don't think it makes much sense. Anyway, this is out of scope of this PR, but will fix that in one of the next ones.